include/net/zzhc.rst

```diff
@@ -0,0 +1,74 @@
+.. _lib_zzhc:
+
+China Telecom ZZHC library
+#################################################
+
+The China Telecom ZZHC library implements the self-registration (Zi-ZHu-Ce, ZZHC) functionality that registers the module information on the server when the network is available.
+
+Overview
+********
+
+The telecommunication modules that are part of the China Telecom network are required to provide certain information stored on the module to the `zzhc.vnet.cn` server.
+This is required for conformity with Chapter 6 of the China Telecom IoT Module Requirements Whitepaper.
+
+The ZZHC library is started as a background thread on boot and it automatically detects the availability of the operator network.
+Once the Internet connectivity is ready and a new SIM card is detected on boot, the library proactively uploads the following information to the ZZHC server:
+- IMEI
+- model number (that is, nRF9160)
+- modem revision
+- nRF9160 SIM ICCID
+- nRF9160 SIM IMSI
+On any failure, which include unknown/negative response from server or any error during program execution, the ZZHC library retries the next hour for at most 5 times.
+
+.. _lib_zzhc_dependencies:
+
+Dependencies
+************
+
+The ZZHC library only when CONFIG_TRUSTED_EXECUTION_NONSECURE is enabled (i.e. Trusted Execution: Non-Secure firmware image).
+
+The ZZHC library requires the following to operate, therefore they are automatically selected when ZZHC library is enabled.
+   * CONFIG_BSD_LIBRARY=y
+   * CONFIG_SETTINGS=y
+   * CONFIG_AT_CMD=y
+   * CONFIG_AT_CMD_PARSER=y
+   * CONFIG_AT_NOTIF=y
+   * CONFIG_BASE64=y
+   * CONFIG_JSON_LIBRARY=y
+
+.. _lib_zzhc_configuration:
+
+Configurations
+**************
+
+The ZZHC library is by default not used in any sample.
+To enable the library so that the end product can be approved by China Telecom, enable Kconfig option CONFIG_ZZHC.
+In prj.conf, set CONFIG_ZZHC=y.
+
+The thread behavior can also be configured using Kconfig options:
+   * To configure the stack size for the thread, change CONFIG_ZZHC_STACK_SIZE.
+   * To configure the thread priority, change CONFIG_ZZHC_THREAD_PRIO.
+
+The minimal heap size required is 4096 bytes.
```

Comment:
Does this heap size requirement refer to the thread behavior options in line 49-50 or to enabling the library in general?

Reply:
No, heap size is irrelevant to thread behavior.

Thread configurations govern how the thread executes. As a naive example, execute 50 lines, then sleep 10ms, then continue execution...etc.

Heap size governs how the library allocate and free memory. If the size is too small, the thread simply kills itself because of memory fault. If it is too large, then it will be a waste of memory.

Reply:
Besides, stack size is not supposed to be touched unless the SDK user need to optimize and push to limits.

Heap size need to be accumulated to the heap size requirements of user application. For example, if the user application requires 16KB heap to run, then to enable zzhc library the SDK user need to increase it to 20KB.